### ISSUE TITLE
Feedback on v1.6

### ISSUE THREAD

**Issue by aubergine10** [dymanoid/RealTime#65]
The 'Wakeup and go to sleep' feature is working well, but after some play testing I have following feedback:

1. Saturdays and Sundays feel like week days
   * Saturday: People will generally get up a little later than usual
   * Sunday: Like Saturday, people will get up later than usual, but they'll also be in bad much earlier
   * Suggestions:
        1. Slider to set `Weekend wake up time`
        2. Slider to set `Sunday sleep time` (Saturday sleep time = weekday sleep time)
2. I yearn for half day during week
    * The city feels really weird without it
    * ...because RealTime is making all other aspects of my city feel more real, which exacerbates discrepancies
    * Suggestions:
        1. Drop-down to choose `Half-day week day` (defaults to "Disabled")
        2. `Start of lunch hour` (existing setting) defines time at which half-day kicks in on selected day of week
        3. `Half-day shutdown amount` option (probability tab) defines how many buildings (ideally) or cims will go home at lunch time on the half-day.
3. Sometimes I want to 'skip to night speed' -- discussed below...

## Skip to night speed...

I'm finding that at a certain point of day (usually late evening), I'm wanting to shift to faster rate of time rather than waiting for sleep time to kick in. I can do this manually, by going in to mod options and increasing day speed, but it's obviously cumbersome to keep doing that.

It's as if my brain is saying "I'm done for the day, let's see how those changes play out, and I'll continue in the morning". So I'm yearning for a "**done for the day"** button, which would instantly trigger night speed setting prior to reaching sleep time, but would not trigger night budgets (they remain in day mode until sleep time is reached).

**Comment by aubergine10** [comment]
Additional feedback...

I've set sleep time to 11pm:

* If camera is zoomed out, not much changes visually
    * could buildings start turning their lights out (particularly residential, but would affect industry and commerce too)?
    * The number of buildings going dark could be guided by the workshift ratios, at least for commerical, industrial and offices.
    * I'd expect most if not all residential to be dark within few hours of sleep time (some high density might stay lit a bit longer)
    * Lights out should be staggered - maybe use `overtime` setting to determine length of transition period?

* Water and Sewage production remains largely unaffected
    * If people are sleeping, why so much water/sewage?
   * After sleep time, could it change in line with the active work shift ratios?

* There's still lots of people out at sleep time, particularly in parks, etc.
    * This means my transport shifts in to low budget while cims are still out partying
    * Would it be possible to have percentage of cims (defined by `main shift` percentage?) choose to head home early so they get home around sleep time?

* I've noticed lots of pensioners partying late in to the night
    * Would it be possible to send them home between 9pm and 11pm regardless of any other settings?

Sunrise:

I've noticed that in slow time (eg. speed set to 1 or 2), the sunrise looks odd, like some alien world with no atmosphere...

![cities_ skylines 15_07_2018 04_56_26](https://user-images.githubusercontent.com/1386719/42730804-2887b816-87f6-11e8-8bcd-31a0bd284e0f.png)

I'm not sure if that's something to do with LUT or map theme or something else, but it looks very odd.

This issue doesn't seem to affect sunsets, I've only seen this sort of thing during sunrise when speed is set really slow.

**Comment by aubergine10** [comment]
## Outside connections

In sleep time, I'd expect traffic from outside road and rail connections to change...

* Fewer passenger trains
    * half usual rate? 
    * could potentially be achieved by turning off rail buildings at night (like service buildings idea mentioned above)?
* More freight trains
    * taking advantage of quiet rail network to import/export goods
* Fewer immigrants, tourists via roads
    * because most people are sleeping
* More goods/industry vehicles
    * taking advantage of less congestion at night for more efficient deliveries
* Less dummy traffic
    * industry and goods vehicles might increase though

**Comment by dymanoid** [comment]
When you post so much separate questions in one issue, I am not able to address them in code. So we will keep this issue as a discussion thread only.

**Comment by dymanoid** [comment]
> Saturdays and Sundays feel like week days

Low priority feature from my point of view. You can post it as a separate issue if you wish.

> Half-day week day

A *very* specialized one. E.g., in Russia, there is no such thing, so the players from there won't need it. In Germany too. I would choose a "no over-complication" way here. So we drop it.

> Skip to night speed

You craved for the "decoupled night & sun". Now you say "gimme that back" (pretending that "a certain point" you mentioned is the sunset time - it's conceptually the same). And again, we have **Real Time** here, not a turn-based strategy ("I'm finished with my turn, let's advance to the next one"). Verdict: this won't be implemented.

**Comment by dymanoid** [comment]
> If camera is zoomed out, not much changes visually

This is something for a separate mod. It changes the rendering/visuals, it's out of scope for **Real Time**.

> Water and Sewage production remains largely unaffected

This is something for another mod. There are other mods that change this (e.g. "Realistic Consumption" etc.), so I don't want to make **Real Time** incompatible with all those.

> There's still lots of people out at sleep time, particularly in parks, etc.

Lots of? There should be significantly less people that go out at night. A screenshot would be helpful.

> I've noticed lots of pensioners partying late in to the night

Probably, this can't be fixed. The game simulates citizen's decisions not for all citizens at the same time, but gradually. Let's say it's 8 p.m. and a pensioner decides to walk in a park. It's ok, because pensioners should go home at 9 p.m. But the next time the game checks this pensioner is maybe 11 p.m. or even midnight or later (this depends on the population size). And of course, that poor pensioner will be there until the next decision says "go home now".

**Comment by dymanoid** [comment]
@aubergine10, any feedback from your side to my comments? I'm open for a discussion if you disagree with any of my replies.

**Comment by aubergine10** [comment]
Yup, sorry for delayed response - been reading up on modding API and might have a go at making mod that alters outside connection traffic based on time of day. Is there some way for mods to work out when the game switches to night budget (ie. transition to sleep time) or inspect RT mod settings?

As for people being out later than expecting, I hadn't considered that the game processes sims gradually over time. Maybe in the hour or two preceding bed time, an increasing random percentage of cims could decide to stay home? It might result in, visually at least, crowds appearing to dissipate towards bed time.

Random thought: How does bed time affect things like night tours on park areas?

Also parks (I've not checked for a few weeks): Do the fireworks still last the whole night? That made sense in vanilla game, because everything was complete chaos. But in RealTime, it would be odd.

Regarding buildings going dark and/or closing after sleep time, IMO that should be within scope of RT - despite it being purely visual thing. RT changes stuff in the simulation, but end users rely on visual feedback to know what's happening (eg. huge traffic jam let's me know it's rush hour).

Put another way: Vanilla game is like an over-blended soup, and I'm like "uhm, where's the croutons?". RealTime adds a bunch of croutons. But there's no point adding croutons if you can't taste them, or, in the case of the game, see them. I've configured when my 'sleep time' crouton kicks in, but it's missing a major visual element when it does: I'm expecting building lights to start going out, and smaller (ideally: those chosen by me via checkbox on their options panel) service buildings to close. In terms of visual feedback to the user about what's happening in the simulation, the croutons have gone soggy in the soup. :)

**Comment by aubergine10** [comment]
From mods' description page:

> Are you wondering why your Cims never sleep and keep working and studying all the night long?

Their circadian rhythms are all messed up due to the lights being on all night :P

**Comment by dymanoid** [comment]
> Is there some way for mods to work out when the game switches to night budget (ie. transition to sleep time)

Yes, the game's `SimulationManager.instance.m_isNightTime` field indicates that. Without **Real Time**, this field depends on the sun and on the 'Use day/night cycle' option (if deactivated, the field is always `false`). With **Real Time**, this field depends on wakeup & go to sleep time.

**Comment by dymanoid** [comment]
> Maybe in the hour or two preceding bed time, an increasing random percentage of cims could decide to stay home?

This logic will be slightly changed in #57 and should become more realistic.

> How does bed time affect things like night tours on park areas?

Bed time --> drastically less visitors everywhere. No special handling for the tours currently.

> Do the fireworks still last the whole night?

Fireworks? Didn't know they are available. So maybe we need a feature to control them.

> buildings going dark and/or closing after sleep time

This is a tough one. I'm not an asset editor and have no idea how to control all that 3D stuff. So of course we can list this feature here, but I doubt I can implement any time soon. A help from some asset editor / Unity3D geek would be highly appreciated.

> closing small local services

I suppose it will be better not to suppress the "disabled" bubble. Otherwise, there's no way for the players to quickly know what is closed at night.

**Comment by aubergine10** [comment]
> Bed time --> drastically less visitors everywhere. No special handling for the tours currently.

It's an odd policy. I'll set some parks up for night tours to see what happens (I don't even know what happens in vanilla game in this context).

> Fireworks? Didn't know they are available.

Feature of parklife park areas, they emanate from main gate on Saturdays (I think?) after sunset and, in vanilla, literally last until sunrise the next day. There's a park policy that makes them happen every night, and one that stops them completely.

Based on parks in UK, I'd expect fireworks to start ~1 hour prior to the park closing for the night (so in RT that would be 1 hour prior to sleep time). But they also always happen at night, so if sleep time happens prior to sunset in mid-summer... 

> I suppose it will be better not to suppress the "disabled" bubble. Otherwise, there's no way for the players to quickly know what is closed at night.

Agreed.

Would it be possible to show alternate bubble so players can visually differentiate between sleeping vs. manually turned off? Something like a ["Zzzz" sleep sign](https://www.colourbox.com/vector/sleep-sign-icon-moon-with-zzz-button-vector-11070432), ideally with dark background so as not to look ugly at night.

**Comment by Sipke82** [comment]
> I suppose it will be better not to suppress the "disabled" bubble. Otherwise, there's no way for the players to quickly know what is closed at night.

Well if the lights are turned off.... ? and isn't lights on/off some boolean somewhere? (sometimes i spot a building with lights off at night)

With regard of turning off, it would be nice if the industry would be turned off (not operating) for like 80% or something (only the night shifts are working!).  I suppose that, if their status is "not operating"they won't spawn trucks either? which would really help t the "quiet night" effect.
Also, i think the Shop zones are still a bit too busy in the middle of night. Same with Parks, i monitored (with traffic monitor) a small park in my highly attractive leisure area, started at around 7 pm around 400 path's towards it. at 1 AM in the middle of the night, still 280 paths directed to it!! (should be way way lower IMO)

My leisure district is really really popular, every night of the week + the entire weekend! we got people partying all night/day long, (also because of late arrival due to lack of parking spaces i think). Still i think it should be really popular on Friday and Saturday night and less popular on the other moments.
_sidenote: the leisure area is in an attractive area of the city :-)_

Perhaps we should think about how do we define weekend? perhaps Friday 5 pm till Sunday 5 pm should be considered weekend, if that is possible? 
On the other days of the week, leisure attractiveness could be turned down? perhaps by putting a certain % on "not operating"? Perhaps that percentage can be based on the work shifts?

And finally shopping, it feels for me that the shopping areas are always a bit under crowded, as there are almost more workers around than shoppers :-) 
But i should observe this a bit closer in the weekends, but that might take me a few days, considering the slow speed i'm playing this game now :-)

Although, this might sounded a bit like a rant, it certainly isn't! I really love this mod and im looking forward to all the new cool stuff!

**Comment by aubergine10** [comment]
> Spike82: Well if the lights are turned off.... ?

"Off" buildings won't light up at night (same in vanilla). But just "lights off" isn't sufficient visualisation, because...

> Spike82: lights on/off some boolean somewhere? (sometimes i spot a building with lights off at night

Yes, vanilla game seems to randomise which buildings have lights on off, but it's generally ineffective because it doesn't seem to factor in zoning type, day of week, time of day, etc.

_So lights out = cims sleeping (eg. residential areas) - or - building turned off at night._

> Spike82: With regard of turning off, it would be nice if the industry would be turned off (not operating) for like 80% or something (only the night shifts are working!).

Could that be part of [Work Hours for Industries](https://github.com/dymanoid/RealTime/issues/68)?

Personally, I'd like to see it work like this:

* Residential: A percentage of the buildings go dark at night, starting an hour or so after sleep time or sunset, whichever occurs latest (cims sleeping)
* Everything else: A checkbox on building info panel determines whether it shuts down for the night 

> Spike82: Perhaps we should think about how do we define weekend? Friday 5 pm till Sunday 5 pm

Yup, that would be nice.  During that period leisure and tourism could get a boost, at other times they'd be reduced?

I to have noticed that my park areas are insanely popular all day. My city is only at 1.5k pop currently!  Doing some checking a lot of unemployed, pensioner and tourist people visit them.

I'm not sure unemployed and pensioners visit parks that regularly - but that sort of tweak is likely for one of the rebalance mods (just guessing) rather than RT?

> Spike82: And finally shopping, it feels for me that the shopping areas are always a bit under crowded, as there are almost more workers around than shoppers

I agree.

I'd expect the pensioners and unemployed people I mentioned above to visit shops more than they visit leisure.

I had considered that it might just be a visual thing - because parks the cims are often "outdoors" so you can see them; in shops they disappear when they enter the shop. However, even just looking at traffic levels, there's a lot more cims travelling to/from parks than there are to/from shops. That leads me to suspect that the simulation is weighted towards leisure (which would make sense as people will want to see leisure getting high use with the Park Life DLC).

**Comment by aubergine10** [comment]
More musings on that shops vs. parks thing (no idea if any of this is feasible or even in scope of RT mod)...

Generic Commercial ("Shopping"):
* Should have baseline attendance, primarily in "main shift" hours
* Pensioners in particular want to do shopping at quiet times (during main shift Mon-Fri)
* Peak in weekday evenings, weekends and (if implemented at some point) "sales" eg. January sales
* Highstreet shopping is on the decline, because people are buying online

Organic Commercial ("Eco Shopping"):
* Based on trends I've observed IRL, these shops should be busier than their generic counterparts (increased baseline attendance rather than increased peaks)

Tourism Commercial:
* I'd expect a baseline level of tourism (eg. business travel and day trips) punctuated by influxes at weekends and and holiday seasons
* I'd expect the tourists to actually spend most of their time either in the hotel or at parks/landmarks

Leisure Commercial:
* I'd expect low baseline attendance
* Peak in the evenings (particularly Fridays), Saturday from lunch onwards (particularly in evening), Sunday mornings (people go home early due to work next day)

* Parks (Buildings)
   * I'd expect relatively calm baseline attendance at most times
   * Peaks would be weekends, warm weather, holidays

* Parks (Areas - with main gate defining type of area)
   * I'd expect relatively calm baseline attendance at most times
   * General peaks would be weekends, warm weather, holidays
   * City: Peak at weekends, big peak during hot weather and summer months
   * Amusement areas: Peak at weekends, big peak in summer months, huge peak during school breaks
   * Zoos: Peak during summer months, huge peak during school breaks
   * Nature: Peak during nice weather and summer months, big peak during holidays

Which reminds me to mention: School breaks! Adding a school break feature (maybe as some sort of "event"? that way players could define the start/end date accurately...) would be awesome, because suddenly all those kids who "work at school" during the day get to run amok - and many of their parents will take time off work in that period too. Commerce and parks would take a hammering.

**Comment by Sipke82** [comment]
> aubergine10 : "Off" buildings won't light up at night (same in vanilla). But just "lights off" isn't sufficient visualisation, because...
IMO that is sufficient... Because we have already read in the mod description  that certain industries and buildings are closed in the night. the turning of the lights gives additional confirmation. Furthermore, because a lot of stuff is shutdown, the streets are near empty, which gives the final confirmation that a lot of stuff is shutdown..
I wouldn't like all kinds of icons all over the city that buildings are shutdown because of night..

**Comment by dymanoid** [comment]
@aubergine10, @Sipke82, After introducing of the personal schedules implemented with #57, the citizens behavior concerning visit place selection is drastically changed. Do we still need this thread? Maybe extract the feature requests I overlooked (if any) and close it?

**Comment by Sipke82** [comment]
I think this thread can be closed, quick review of the thread didn't spot more feature requests which aren't already listed.

**Comment by aubergine10** [comment]
I'll split out the few remaining items and then close this ticket shortly...
